nexus/db-queries/src/db/datastore/deployment.rs

```diff
@@ -210,7 +210,7 @@ impl DataStore {
             })
             .await
             .map_err(|e| match err.take() {
-                Some(txn_error) => txn_error.into(),
+                Some(txn_error) => txn_error.into_public_ignore_retries(),
```

Comment:
There are several spots "immediately after a transaction" where this conversion is valid.

These have been changed from "into()" -> "into_public_ignore_retries()", but are functionally identical.